applications/IgaApplication/tests/shell_3p_element_tests.py

```diff
@@ -0,0 +1,238 @@
+#   KRATOS  _____________
+#          /  _/ ____/   |
+#          / // / __/ /| |
+#        _/ // /_/ / ___ |
+#       /___/\____/_/  |_| Application
+
+import KratosMultiphysics as KM
+import KratosMultiphysics.StructuralMechanicsApplication as SMA
+import KratosMultiphysics.python_linear_solver_factory as linear_solver_factory
+
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import os
+
+
+def GetFilePath(fileName):
+    return os.path.dirname(__file__) + "/" + fileName
+
+
+class Shell3pElementTests(KratosUnittest.TestCase):
+
+    def solve_cantilever(create_geometry):
+        model = KM.Model()
+        model_part = model.CreateModelPart('Model')
+
+        model_part.AddNodalSolutionStepVariable(KM.DISPLACEMENT)
+        model_part.AddNodalSolutionStepVariable(KM.REACTION)
+        model_part.AddNodalSolutionStepVariable(SMA.POINT_LOAD)
+
+        # create property for shell elements
+
+        shell_properties = model_part.GetProperties()[1]
+        shell_properties.SetValue(KM.THICKNESS, 0.1)
+        shell_properties.SetValue(KM.YOUNG_MODULUS, 200000000)
+        shell_properties.SetValue(KM.POISSON_RATIO, 0)
+        shell_properties.SetValue(KM.CONSTITUTIVE_LAW, SMA.LinearElasticPlaneStress2DLaw())
+
+        # create a nurbs surface
+        surface = create_geometry(model_part)
+
+        # create quadrature_point_geometries
+        quadrature_point_geometries = KM.GeometriesVector()
+
+        surface.CreateQuadraturePointGeometries(quadrature_point_geometries, 3)
+
+        element_id = 1
+        for i in range(0, len(quadrature_point_geometries)):
```

Comment:
I am not familiar with the Kratos vector but would

```python
for quadrature_point_geometry in quadrature_point_geometries:
    model_part.CreateNewElement('Shell3pElement', element_id, quadrature_point_geometry, shell_properties)
```

work?

Reply:
Unfortunately, that isn't working. I've even explicitly checked it. 

The problem is that the vector counter-part is a quite weird vector with an access combination of member and pointer. Probably, one could check the correct export of the vector, but I want to keep that for future research ;).

Thanks for suggesting it, though.